hadoop-project/pom.xml

```diff
@@ -1727,6 +1728,11 @@
         <artifactId>jna</artifactId>
         <version>${jna.version}</version>
       </dependency>
+      <dependency>
```

Comment:
Looks okay to me https://mvnrepository.com/artifact/io.airlift/aircompressor/0.16
No compile time dependency. A shaded Hadoop 2 jar in provided dependency.


What's tricky is it has test scope dependency on jmh-core, which is GPL 2.0.

Reply:
Since we don't depend on the test scope, I think we should be fine.

Reply:
@steveloughran we would like to actually bundle this jar into common since this is a very clean jar and used in many projects such as presto and orc to provide their compression codecs. 

In fact, we would like to have snappy to fallback to aircompressor if no native lib is provided. For many of our devs, it's non-trivial to setup native libs in their development env since it requires to compile many native libs from sources and install them into LD path. If we can fallback to pure java snappy implementation in aircompressor, it will make the developers' life a way easier.

Reply:
My point was that aircompressor has non-optional dependency on GPL artifact and thus making it GPL as well. 
That said, I just verified mjh-core is not in the Hadoop dependency after the patch. 
We should talk to aircompressor developers to clarify the license issue.

Reply:
I would like to mention that other apache projects such as Presto and Spark all include this as dep, so would be great to get the clarification since it's heavily used in Apache community..